JobScheduler.js

```diff
@@ -147,26 +148,36 @@ class JobScheduler {
-              logger.warn(`System in maintenance since ${maintenanceInfo.createdAt}. Exceeds configured maintenance timeout. Flagging the current maintenance window as aborted.`);
-              const success = resolve;
-              return maintenanceManager
-                .updateMaintenace(`System in maintenance beyond configured timeout time ${config.scheduler.maintenance_mode_time_out/1000/60} (mins)`,
-                  CONST.OPERATION.ABORTED,
-                  CONST.SYSTEM_USER)
-                .then(success)
-                .catch((err) => {
-                  logger.error('error occurred while updating maintenance info', err);
-                  reject(err);
-                })
-                .finally(() => {
-                  if (this.intervalTimer) {
+            const maintInfoAttrs = ['progress', 'state', 'completedAt', 'reason', 'toVersion', 'fromVersion', 'createdAt', 'updatedAt', 'createdBy', 'updatedBy'];
+            if (_.get(maintenanceInfo, 'state', '') !== CONST.OPERATION.IN_PROGRESS) {
+              logger.info(`+-> System is not in maintenance, but its current state is: ${_.get(maintenanceInfo, 'state', '')}, not as expected. `);
+              logger.info('checking if service fabrik is up, inspite of unexpected maintenance state - ', _.pick(maintenanceInfo, maintInfoAttrs));
+              this.isServiceFabrikUp()
+                .then(status => {
+                  logger.info(`SF Connected to DB :- ${status}`);
+                  if (status === true) {
+                    logger.info('Service fabrik is up, going ahead with starting the scheduler.');
                     clearInterval(this.intervalTimer);
+                    return resolve();
                   }
                 });
+            } else {
+              logger.info('+-> System is in maintenance :', _.pick(maintenanceInfo, maintInfoAttrs));
+              const currTime = moment();
+              const maintenanceStartTime = _.get(maintenanceInfo, 'createdAt');
+              if (maintenanceStartTime && currTime.diff(maintenanceStartTime) > config.scheduler.maintenance_mode_time_out) {
+                logger.warn(`System in maintenance since ${maintenanceInfo.createdAt}. Exceeds configured maintenance timeout  ${config.scheduler.maintenance_mode_time_out} (ms). Flagging the current maintenance window as aborted.`);
+                maintenanceManager
+                  .updateMaintenace(`System in maintenance beyond configured timeout time ${config.scheduler.maintenance_mode_time_out/1000/60} (mins). JobScheduler aborting it.`,
+                    CONST.OPERATION.ABORTED,
+                    CONST.SYSTEM_USER)
+                  .then(() => logger.info('JobScheduler Aborted current maintenance window'))
+                  .catch((err) => {
+                    logger.error('error occurred while aborting maintenance - ', err);
```

Comment:
all logger statement initiate with Upper case.

Reply:
This is personal perference and there is no guideline perse as code base has many loggers like the above. Will incorporate the above, if the the same is agreed upon and enforced as a guideline.